### PR TITLE
TF-38711 Add Stack Support For Notification Configurations

### PR DESCRIPTION
**Pull request by Latuang**
## Description

Adds Stack as a valid subscribable type for v1 notification configurations. It extends NotificationConfigurationSubscribableChoice and updates routing and validation to support the stacks/{id}/notification-configurations endpoint, following the existing team/project pattern.

This is step 1 of TF-38711; step 2 will add support in terraform-provider-tfe so customers can manage stack notifications through Terraform.

## PCI review checklist

- [x] I have documented a clear reason for, and description of, the change I am making.

- [ ] If applicable, I've documented a plan to revert these changes if they require more than reverting the pull request.

- [ ] If applicable, I've documented the impact of any changes to security controls.

  Examples of changes to security controls include using new access control methods, adding or removing logging pipelines, etc.

### CODE REVIEW

File: v1.go
Comment (jillirami) [inline v1.go]:
```suggestion
	// Required: The workspace, team, project, or stack that the notification configuration is associated with.
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# v2.7.0
+
+* Enhancement: Add support for stack-level notification configurations. `NotificationConfigurationSubscribableChoice` now accepts a `Stack` field, allowing notification configurations to be created and listed for stacks via `stacks/{stack_id}/notification-configurations`.
```
Comment (jillirami) [inline CHANGELOG.md]: This PR hooks into existing support for notification_configurations with varying subscribable types ([reference PR](https://github.com/hashicorp/go-tfe/pull/1350)) for all CRUD operations (read, update, delete included) to be supported. This would be accurate with:

```suggestion
* Enhancement: Add support for stack-level notification configurations. `NotificationConfigurationSubscribableChoice` now accepts a `Stack` field, supporting notification configuration CRUD operations.
```

**Review by jillirami — COMMENTED** [review]
It would be ideal to have tests with this code change and not rely only on terraform-provider-tfe tests, however it seems things have changed in this repository since I last [wrote tests](https://github.com/hashicorp/go-tfe/pull/1350/changes). I defer to the core-cloud team to understand the current best practices for testing. No blocking comments 👍

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# v2.7.0
```
Comment (shwetamurali) [inline CHANGELOG.md:11]: This branch conflicts with main, which already contains the v2.7.0 and v2.8.0 release sections. Please rebase and move this entry under the current release section!

**Comment by shwetamurali** [comment]
Could we add focused tests for the new stack subscribable behavior? At minimum, these should cover list/create routing to `stacks/{id}/notification-configurations`, JSON:API encoding/decoding of the `stacks` relationship, and `ErrInvalidStackID`.

**Review by ctrombley — APPROVED** [review]
Looks great, thanks @Latuang!
